src/modules/RootPart.rb

```diff
@@ -291,8 +290,8 @@ def AddMountedPartition(partinfo)
 
     # Check the filesystem of a partition.
     def FSCKPartition(partition)
-      detected_fs = probed.partitions.with(name: partition).filesystems.first.to_s.to_sym
-      if detected_fs == :ext2
+      detected_fs = fstype_for_device(probed, partition)
+      if detected_fs == "ext2"
```

Comment:
Wouldn't it be better if `fstype_for_device` would return the `Filesystems::Type` object so we could do things like this?
```ruby
if detected_fs.is?(:ext2)
```

That assuming, of course, that so far we don't want to fix the root of the problem (that is, `FSCKPartition` should receive a partition object and not a string).